project/yui0303/README.md

```diff
@@ -0,0 +1,225 @@
+
+Project URL: [https://github.com/yui0303/Matrix-Diagonalization](https://github.com/yui0303/Matrix-Diagonalization)
+
+
+## Problem to Solve
+
+There a serverl steps to do to diagonalize a square matrix.  
+1. Calcualete the eigenvalues of the matrix.
+2. Find the corresponding eigenvectors of eigenvalues.
+3. Check whether a matrix is diagonalizable.
+
+
+## How to calculate the eignvaluses and the corresponding eigenvectors
+### Definition
+
+On $A_{n\times n}$
+- eigenvalue $(\lambda)$: It is a **scalar**. Exist a vector $(v \neq \vec{0})$ such that $Av = \lambda v$.
+- eigenvector $(v)$: Non-zero vector $(v \neq \vec{0})$. Exist a scalar ($\lambda$) such that $Av = \lambda v$.
+
+Note that: eigenvector cannot be $\vec{0}$, but eigenvalue can be 0.
+
+### Computaion
+
+- eigenvalue
+    - Assume c is an eigenvalue of matrix A $\iff$ $det(A-cI) = char_{A}(c) = 0$  
+    - $det(A)$ is the determinant of matrix $A$ and $char_A(x)$ is the characteristic polynomial of matrix $A$.  
+- eigenvector
+  - If $\lambda$ is a eigenvalue of the matrix $A$, then $V(\lambda) = \lbrace v \in V \parallel Av=\lambda v \rbrace = ker(A-\lambda I)$ is the eigenspace of $\lambda$. Note that $Ker(A)$ is the kernel space of matrix $A$.
+  - Eigenvector is the non-zero vector in $V(\lambda)$.  
+
+</br>
+
+e.g.  
+
+$$A=
+\begin{bmatrix}
+  1 & 4 \\
+  5 & 0
+\end{bmatrix}
+$$
+
+$$\therefore det(A-\lambda I) = 0 \Rightarrow (1-\lambda)(-\lambda) -20 = 0$$ 
+$$\therefore \lambda = 5, -4$$  
+
+$$V(5)=Ker(
+\begin{bmatrix} 
+  -4 & 4 \\ 
+  5 & -5
+\end{bmatrix}
+)
+= span(
+\lbrace
+\begin{bmatrix} 
+  1 \\ 
+  1 
+\end{bmatrix}
+\rbrace
+)
+$$  
+
+$$
+V(-4)=Ker(
+\begin{bmatrix} 
+  5 & 4 \\ 
+  5 & 4
+\end{bmatrix}) 
+= span(
+\lbrace
+\begin{bmatrix} 
+  4 \\
+  -5 
+\end{bmatrix}
+\rbrace
+)
+$$
+
+Therefore, we can get the
+
+$$
+\begin{bmatrix} 
+  1 \\ 
+  1 
+\end{bmatrix}
+$$
+
+and 
+
+$$
+\begin{bmatrix} 
+4 \\
+-5 
+\end{bmatrix}
+$$ 
+
+as eigenvectors respectively.  
+
+### Challenges
+1. Solve the characteristic polynomial.
+   - Maybe use QR Decomposition to find the eigenvalues
+2. Solve the Kernel space.
+   - Using Gaussian Elimination
+   - Other Methods?
+
+
+## How to diagonalize the matrix
+
+On $A_{n\times n}$ ,  
+If the eignvalues of $A$ is $\lambda_{1}, \lambda_{2}, ..., \lambda_{n}$ and the corresponding eigenvectors are $v_1, v_2, ..., v_n$, then the diagonalization of $A$ is shown as below.  
+
+$$
+P^{-1}AP=
+\begin{bmatrix}
+\lambda_{1} & & & & \\
+ & \lambda_{2} & & &\\
+ & & . & & \\
+ & & & . & \\
+ & & & & \lambda_{n}
+\end{bmatrix} 
+,
+P=
+\begin{bmatrix}
+  v_1 & v_2 & . & . & v_n
+\end{bmatrix}
+$$
+
+**prerequisite: A has n independent eigenvectors.**
+
+## Prospective Users
+
+The diagonal matrix has the property that all of its off-diagonal elements are zero. There are some prospective usages for related users are shown below.
```

Comment:
Eigenvalue problems are a big subject.  How large are the matrices you want to solve?  Is it dense or sparse?